### DISCUSSION TITLE
Custom dropdown instead of the generic that we get in react query builder

### DISCUSSION THREAD

**Discussion by kuljeet2604** [react-querybuilder/react-querybuilder#629] [Q&A] [answered]
Hi

can we have a Custom dropdown instead of the generic that we get in react query builder.
suppose clicking it (the dropdown) opens a section which as filter or any other selection that component (custom build) will be used to select the attribute/field.
and the custom component will return me the attribute.

basically i wanted to develop a custom dropdown instead of the one provided here

**Comment by kuljeet2604** [comment, marked as answer]
can be done with `fieldSelector` prop

```
controlElements={{
  fieldSelector: FieldSelector,
}}
```

**Reply by jakeboone02** [reply]
Yep, and the [home page](https://react-querybuilder.js.org/) has a example using `react-select`. It's for the `valueEditor` and not `fieldSelector`, but the concept is the same. Relevant code for the home page is [here (component)](https://github.com/react-querybuilder/react-querybuilder/blob/196ed93117a471cafa4079a7d7cc025e5bfca361/website/src/pages/index.tsx#L72-L129) and [here (implementation)](https://github.com/react-querybuilder/react-querybuilder/blob/196ed93117a471cafa4079a7d7cc025e5bfca361/website/src/pages/index.tsx#L252-L257).
